x/ccv/provider/keeper/proposal.go

```diff
@@ -466,73 +447,52 @@ func (k Keeper) BeginBlockCCR(ctx sdk.Context) {
 	k.DeletePendingConsumerRemovalProps(ctx, propsToExecute...)
 }
 
-// ConsumerRemovalPropsToExecute iterates over the pending consumer removal proposals
-// and returns an ordered list of consumer removal proposals to be executed,
-// ie. consumer chains to be stopped and removed from the provider chain.
+// GetConsumerRemovalPropsToExecute returns an ordered list of pending consumer
+// removal proposals that are ready to be executed,
+// i.e., consumer chains to be stopped and removed from the provider chain.
 // A prop is included in the returned list if its proposed stop time has passed.
 //
 // Note: this method is split out from BeginBlockCCR to be easily unit tested.
-func (k Keeper) ConsumerRemovalPropsToExecute(ctx sdk.Context) []types.ConsumerRemovalProposal {
-
-	// store the (to be) executed consumer removal proposals in order
+func (k Keeper) GetConsumerRemovalPropsToExecute(ctx sdk.Context) []types.ConsumerRemovalProposal {
 	propsToExecute := []types.ConsumerRemovalProposal{}
 
-	k.IteratePendingConsumerRemovalProps(ctx, func(stopTime time.Time, prop types.ConsumerRemovalProposal) (stop bool) {
-		if !ctx.BlockTime().Before(stopTime) {
+	k.IteratePendingConsumerRemovalProps(ctx, func(prop types.ConsumerRemovalProposal) (stop bool) {
+		if !ctx.BlockTime().Before(prop.StopTime) {
 			propsToExecute = append(propsToExecute, prop)
 			return false // do not stop the iteration
 		}
-		// No more proposals to check, since they're stored/ordered by timestamp.
-		return true // stop
+		// stop iteration; proposals are ordered by stop time,
+		// so no additional pending props are ready to act upon
+		return true
 	})
 
 	return propsToExecute
 }
 
+// IteratePendingConsumerRemovalProps iterates through the pending consumer removal proposals.
+//
+// Note that the pending consumer removal proposals are stored under keys with the following format:
+// PendingCRPBytePrefix | stopTime | chainID
```

Comment:
opinionated nit: I would generally not put this kind of implementation detail in the function docstring. It's not bad to have it inside though.

Reply:
For now, I would like to have clarity about what's happening inside the iterator. Once we have guidelines of how to use the SDK store, we can drop them. Maybe a cleaner way is to use `cosmos-sdk/store/prefix`, see https://github.com/cosmos/cosmos-sdk/blob/47f46643affd7ec7978329c42bac47275ac7e1cc/x/capability/keeper/keeper.go#L455

Reply:
Imo comments of these type are useful